lib/util/lifted/influx/httpd/handler_metrics.go

```diff
@@ -0,0 +1,263 @@
+	meta2 "github.com/openGemini/openGemini/lib/util/lifted/influx/meta"
+	"github.com/openGemini/openGemini/lib/util/lifted/influx/query"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
+)
+
+var (
+	metricsHandler      http.Handler
+	openGeminiCollector *OpenGeminiCollector
+)
+
+func init() {
+	// Create a custom registry
+	registry := prometheus.NewRegistry()
+	// Instantiate a custom collector
+	openGeminiCollector = NewOpenGeminiCollector()
+	// Register the collector to the registry
+	err := registry.Register(openGeminiCollector)
+	if err != nil {
+		log.Fatal("Failed to register collector:", err)
+	}
+
+	metricsHandler = promhttp.HandlerFor(registry, promhttp.HandlerOpts{Registry: registry})
+}
+
+type OpenGeminiCollector struct {
+	*metrics.BaseCollector
+	indexMap map[string]*metrics.ModuleIndex
+}
+
+func NewOpenGeminiCollector() *OpenGeminiCollector {
+	c := &OpenGeminiCollector{
+		BaseCollector: metrics.NewBaseCollector(metrics.ComIndexRegistry),
+		indexMap:      make(map[string]*metrics.ModuleIndex),
+	}
+	// Initialize metrics
+	for moduleName, index := range c.IndexRegistry {
+		desc := make(map[string]*prometheus.Desc)
+		for indexName, help := range index.HelpMap {
+			desc[fmt.Sprintf("last_%s", indexName)] = metrics.NewDesc("", indexName, help, index.Labels)
+		}
+		c.AllModulesDesc[moduleName] = desc
+	}
+
+	return c
+}
+
+func (c *OpenGeminiCollector) Collect(ch chan<- prometheus.Metric) {
+	var indexValue interface{}
+	indexMap := c.indexMap
+	for moduleName, index := range c.IndexRegistry {
+		value, ok := indexMap[moduleName]
+		if ok {
+			// Get the label of the module
+			labelValues := make([]string, 0)
+			for _, labelName := range index.Labels {
+				labelValue, ok := value.LabelValues[labelName]
+				if !ok {
+					labelValues = append(labelValues, "nil")
+				}
+
+				labelValues = append(labelValues, labelValue)
+			}
+
+			for indexName := range index.HelpMap {
+				indexName = fmt.Sprintf("last_%s", indexName)
+				indexValue, ok = value.MetricsMap[indexName]
+				// If the indicator exists, write the indicator
+				if ok {
+					ch <- prometheus.MustNewConstMetric(c.AllModulesDesc[moduleName][indexName], prometheus.GaugeValue,
+						indexValue.(float64), labelValues...)
+				}
+			}
+		}
+	}
+
+}
+
+func (h *Handler) serveMetrics(w http.ResponseWriter, r *http.Request, user meta2.User) {
+	for moduleName, index := range openGeminiCollector.IndexRegistry {
+		moduleIndex, err := getMetrics(h, r, user, moduleName, index.Labels)
+		if err != nil {
+			continue
+		}
+		openGeminiCollector.indexMap[moduleName] = moduleIndex
+	}
+
+	metricsHandler.ServeHTTP(w, r)
+}
+
+// serverProm
+func getMetrics(h *Handler, r *http.Request, user meta2.User, tableName string, labelNames []string) (*metrics.ModuleIndex, error) {
+	// Retrieve the underlying ResponseWriter or initialize our own.
+	start := time.Now()
```

Comment:
```suggestion
	start := time.Now()
```
No need for "_internal" db.